drivers/ieee802154/ieee802154_mcr20a.c

```diff
@@ -202,12 +202,13 @@ bool _mcr20a_write_burst(struct mcr20a_spi *spi, bool dreg, u16_t addr,
 {
 	bool retval;
 
-	k_sem_take(&spi->spi_sem, K_FOREVER);
-
-	if ((len + 2) > sizeof(spi->cmd_buf)) {
+	if ((len + 2) >= sizeof(spi->cmd_buf)) {
 		SYS_LOG_ERR("Buffer length too large");
```

Comment:
hopefully in a near future we'll be able to get rid of that weird check, using "new" (some months old already..) SPI API.